agent/command.go

```diff
@@ -81,7 +89,8 @@ func (a *Agent) runCommands(ctx context.Context, tc *taskContext, commands []mod
 			}
 		}
 	}
-	return nil
+
+	return errors.WithStack(err)
```

Comment:
Do we need the stack here? The only error that can be returned here is the panic, and we're already logging the stack in the recover